integration/test/performancetest/transmitter.go

```diff
@@ -210,39 +211,56 @@ func (transmitter *TransmitterAPI) Parser(data []byte) (map[string]interface{},
 	
 	for _, rawMetricData := range dataHolder {
 		numDataPoints := float64(len(rawMetricData.Timestamps))
-		var avg float64
-		max := 0.0
-		min := 10000.0
-		if numDataPoints <= 0{
-			avg = 0.0
-			max = 0.0
-			min = 0.0
-			log.Fatalf("Error there is no data points")
-		}else{
-			// @TODO:ADD GetMetricStatistics after merging with data collection code
-			sum :=0.0
-			for _,val := range rawMetricData.Values {
-				sum += val
-				if max < val{
-					max = val
-				}
-				if min > val {
-					min = val
-				}
-			}
-			
-			avg =  sum /numDataPoints
-		}
+		
+		avg, min, max, p99Val, stdDev := CalcStats(rawMetricData.Values)
+		
 		//----------------
 		metric := Metric{
 			Average: avg,
 			Max:     max,
 			Min:     min,
-			P99:     0.0,
-			Std: 	rand.Float64(),
+			P99:     p99Val,
+			Std: 	 stdDev,
 			Period:  int(METRIC_PERIOD / (numDataPoints)),
 			Data:    rawMetricData.Values}
 		packet[rawMetricData.Label] = metric
 	}
 	return packet, nil
+}
+
+//CalcStats takes in an array of data and returns the average, min, max, p99, and stdev of the data
+func CalcStats(data []float64) (float64, float64, float64, float64, float64) {
+	length := len(data)
+	if length == 0 {
+		return -1.0, -1.0, -1.0, -1.0, -1.0
+	}
+	if length < 99 {
+		log.Println("Note: less than 99 values given, p99 value will be equal the max value")
+	}
+
+	//make a copy so we aren't modifying original
+	dataCopy := make([]float64, length)
+	copy(dataCopy, data)
+	sort.Float64s(dataCopy)
+	
+	min := dataCopy[0]
+	max := dataCopy[length - 1]
+	
+	sum := 0.0
+	for _, value := range dataCopy {
+		sum += value
+	}
+
+	avg := sum / float64(length)
+
+	p99Index := length - 1 - (length / 99)
```

Comment:
Interesting. If there are 500 values, then 500/99 would integer divide to 5, right? So we'd sort and then get the 94th index value?

Reply:
500/99 = 5 -> 500 - 1 - (5) = 500 - 6 = 494. But the array is zero-indexed so this would be the 495th value and 99% of 500 is 495.
But I agree that this is kind of convoluted. I will rewrite it to cast to a float, take 99% (length * .99) and then cast back to int and subtract one to get index. More readable that way.

Reply:
Oh that makes more sense. That's my bad.